www/apps/book/app/learn/fundamentals/modules/db-operations/page.mdx

```diff
@@ -683,3 +754,74 @@ class BlogModuleService {
   }
 }
 ```
+
+### Complex Transaction Example
+
+Here's a more practical example demonstrating multiple operations in a single transaction:
+
+```ts
+await productRepository.transaction(async (transactionManager) => {
+  // Create a new product
+  const product = transactionManager.create(Product, {
+    title: "Limited Edition Watch",
+    price: 1299.99
+  })
+  
+  // Update inventory in the same transaction
+  const inventory = await transactionManager.findOne(Inventory, {
+    where: { sku: "WATCH-LE-2023" }
+  })
+  
+  inventory.quantity = inventory.quantity - 1
+  
+  // Save both changes atomically
+  await transactionManager.persistAndFlush([product, inventory])
+  
+  return product
+})
+```
+
+### Advanced SQL Query Examples
+
+Here are examples of more complex SQL queries for common e-commerce needs:
+
+```ts
+// Get popular products with their categories
+async function getPopularProducts() {
+  const knex = this.manager_.getKnex()
+  
+  return await knex.raw(`
+    SELECT 
+      p.id, 
+      p.title, 
+      p.price, 
+      c.name as category_name,
+      COUNT(oi.id) as order_count
+    FROM products p
+    LEFT JOIN product_categories pc ON p.id = pc.product_id
+    LEFT JOIN categories c ON pc.category_id = c.id
+    LEFT JOIN order_items oi ON p.id = oi.product_id
+    GROUP BY p.id, c.id
+    ORDER BY order_count DESC
+    LIMIT 10
+  `)
+}
+
+// Calculate sales metrics
+async function getSalesMetrics() {
+  const knex = this.manager_.getKnex()
+  
+  return await knex.raw(`
+    SELECT 
+      DATE_TRUNC('month', o.created_at) as month,
+      SUM(o.total) as revenue,
+      COUNT(o.id) as orders,
+      COUNT(DISTINCT o.customer_id) as unique_customers
+    FROM orders o
+    WHERE o.status = 'completed'
+    GROUP BY month
+    ORDER BY month DESC
+    LIMIT 12
+  `)
+}
+```
```

Comment:
The practices in these code snippets don't follow our best practices, as they try to create / query data from other modules (which violates module isolation). If you want to add examples, please use the practices recommended in this documentation page.